doc/designs/rbcd.md

```diff
@@ -0,0 +1,385 @@
+|servicedelegationrule-find            |Search for service delegations rule.                        |
+|servicedelegationrule-remove-member   |Remove member from a named service delegation rule.         |
+|servicedelegationrule-remove-target   |Remove target from a named service delegation rule.         |
+|servicedelegationrule-show            |Display information about a named service delegation rule.  |
+|servicedelegationtarget-add           |Create a new service delegation target.                     |
+|servicedelegationtarget-add-member    |Add member to a named service delegation target.            |
+|servicedelegationtarget-del           |Delete service delegation target.                           |
+|servicedelegationtarget-find          |Search for service delegation target.                       |
+|servicedelegationtarget-remove-member |Remove member from a named service delegation target.       |
+|servicedelegationtarget-show          |Display information about a named service delegation target.|
+
+### Resource-based constrained delegation design
+
+Resource-based constrained delegation stores information in the target service
+LDAP entry. This information is represented with `memberPrincipal` attribute
+and is allowed with objectClass `resourceDelegation`. If a Kerberos principal
+is mentioned in the `memberPrincipal` attribute of the LDAP entry and
+objectClass `resourceDelegation` is present in the same entry, KDB driver will
+use information from the `memberPrincipal` attribute to check whether a service
+asking for S4U2Proxy extension is allowed to send a forwarded user ticket to
+this service.
+
+This approach allows implementing MS-SFU's `ServicesAllowedToReceiveForwardedTicketsFrom`
+information.
+
+Management of the resource-based constrained delegation is integrated into `ipa
+host` and `ipa service` commands.
+
+|Command                         |Description                                                 |
+|--------------------------------|------------------------------------------------------------|
+|service-add-delegation          |Add new resource delegation to a service                    |
+|service-allow-add-delegation    |Allow users, groups, hosts or host groups to handle a resource delegation of this service.|
+|service-disallow-add-delegation |Disallow users, groups, hosts or host groups to handle a resource delegation of this service.|
+|service-remove-delegation       |Remove resource delegation from a service|
+|host-add-delegation             |Add new resource delegation to a host|
+|host-allow-add-delegation       |Allow users, groups, hosts or host groups to handle a resource delegation of this host.|
+|host-disallow-add-delegation    |Disallow users, groups, hosts or host groups to handle a resource delegation of this host.|
+|host-remove-delegation          |Remove resource delegation from a host|
```

Comment:
I don't understand the disallow delegation. Shouldn't this be the default unless one has been explicitly defined?

Reply:
The allow-add/disallow-add pairs work together. Once you  added permission with allow command, you can remove it with disallow. That's the point of these commands. The default is that only manager of the resource can handle it.
I think I need to extend the documentation to say that explicitly.

Reply:
Extended documentation.

Reply:
So  host-[dis]allow-add-delegation is a boolean? I think what would really help would be some example usages in case I'm just being thick.